### PR TITLE
Update histogram to gauge estimation

### PR DESCRIPTION
**Pull request by pichlermarc**
This PR updates the histogram to gauge estimation so that the statement `min <= mean <= max` holds for all histograms that are converted to gauges. 
Also updates the `README.md` to mention that histogram estimation is performed.

### CODE REVIEW

File: src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricsExtensions.cs [hunk 1]
```diff
@@ -107,17 +108,12 @@ private static double GetMinFromBoundaries(double histogramSum, long histogramVa
...
+						// point. However, this might lead to a min bigger than the mean, thus choose the minimum
+						// of the following:
 						// - The lowest boundary
-						// - The sum (smallest if there are multiple negative measurements smaller than the lowest
-						// boundary)
-						// - The average in the bucket (smallest if there are multiple positive measurements
-						// smaller than the lowest boundary)
+						// - The average of the histogram (histogram sum / sum of counts)
 						return Math.Min(
 							Math.Min(buckets[i].ExplicitBound, histogramSum),
```
Comment (pirgeo) [inline src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricsExtensions.cs:118]: I dont think we need this anymore, right? We just use the min of bound and mean.

Comment (pirgeo) [reply src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricsExtensions.cs:118]:
```suggestion
							buckets[i].ExplicitBound, 
```

**Review by pirgeo — COMMENTED** [review]
Thanks for updating the tests. I think there is a small bug still!

File: src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricsExtensions.cs (hunk 1 above)
Comment (pichlermarc) [reply src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricsExtensions.cs]: Thanks for catching this, I missed that one. Fixed in [510999b](https://github.com/dynatrace-oss/opentelemetry-metric-dotnet/pull/32/commits/510999bd24f4058c72db3e9950bbc6a3827abfff).

**Review by pirgeo — APPROVED** [review]
Thanks for also making the tests more readable!

File: src/Dynatrace.OpenTelemetry.Exporter.Metrics.Tests/DynatraceMetricsExporterTests.cs [hunk 2]
```diff
@@ -662,30 +672,184 @@ public async Task Export_Histogram_CustomBounds()
 			AssertExportRequest(actualRequestMessage);
 		}
 
+		public static IEnumerable<object[]> MinMaxEstimationData =>
```
Comment (joaopgrassi) [inline src/Dynatrace.OpenTelemetry.Exporter.Metrics.Tests/DynatraceMetricsExporterTests.cs:675]: Nice :) 

I have one nit only: The test data is quite long and it kinda gets in the way of reading the tests. We could put in its own class, like the examples here https://andrewlock.net/creating-parameterised-tests-in-xunit-with-inlinedata-classdata-and-memberdata/#loading-data-from-a-method-on-the-test-class

After we are on .NET 6, the new type `HistogramTestData` could even become a record type, that makes it even shorter :)

Comment (pichlermarc) [reply src/Dynatrace.OpenTelemetry.Exporter.Metrics.Tests/DynatraceMetricsExporterTests.cs]: Great idea :slightly_smiling_face: 
Done in [891a873](https://github.com/dynatrace-oss/opentelemetry-metric-dotnet/pull/32/commits/891a873fe94b90877f5d85b4a24b1b0a16b8855b)!

File: src/Dynatrace.OpenTelemetry.Exporter.Metrics.Tests/DynatraceMetricsExporterTests.cs [hunk 3]
```diff
@@ -33,6 +33,16 @@
 
 namespace Dynatrace.OpenTelemetry.Exporter.Metrics.Tests
 {
+	public sealed class HistogramTestData
```
Comment (joaopgrassi) [inline src/Dynatrace.OpenTelemetry.Exporter.Metrics.Tests/DynatraceMetricsExporterTests.cs:36]: Nit: We could place this inside the `MinMaxEstimationData` class, to be close to where it's instantiated. Also fine as is, your call :)

**Review by joaopgrassi — APPROVED** [review]

File: src/Dynatrace.OpenTelemetry.Exporter.Metrics.Tests/DynatraceMetricsExporterTests.cs (hunk 3 above)
Comment (pichlermarc) [reply src/Dynatrace.OpenTelemetry.Exporter.Metrics.Tests/DynatraceMetricsExporterTests.cs]: Done in [0d7b371](https://github.com/dynatrace-oss/opentelemetry-metric-dotnet/pull/32/commits/0d7b371c7664476ac2c55fe0ff29c1465684e471). :slightly_smiling_face:

**Pull request merged**